### DISCUSSION TITLE
Slow FCP (first contentful paint) of 0.5s

### DISCUSSION THREAD

**Discussion by johannesschaffer** [remix-run/remix#5597] [Q&A]
I just bootstrapped an empty remix project (without any template) & used the App server. Built in production mode & served locally from my (very fast) PC, lighthouse reports a FCP of 0.5s. Next.js reaches 0.2s even in dev mode. Is this normal? Can I somehow improve that?

**Comment by wcjohnson11** [comment]
Similarly interested here.

I ran into this with an app with several routes and I thought it was all of the routes loaders running in parallel blocking but with a fresh app I see the same thing. After the first load, the site is blazing fast but that first load before anything is cached seems to start with a FCP of ~0.5s and once I add some other code to the root loader it starts to range from 0.6s to 0.9s.

So far I've been testing on Vercel and will test elsewhere but is there some best practices for improving FCP of an index/root page at least?

**Comment by CodingMeSwiftly** [comment]
+1 also interested in that. I followed the official Remix [tutorial](https://remix.run/docs/en/main/start/tutorial) and ran lighthouse after the last step. FCP was ~2s for mobile, 0.5s for desktop

**Comment by luksch42** [comment]
Any news on this? Can't sleep well without reaching the 100 performance points on lighthouse 🤡

**Comment by ClimenteA** [comment]
Same here: ~65 performance on lighthouse. A simple website 2-3 links 2 toggles.

**Reply by kiliman** [reply]
Kind of hard to say without seeing the code, but that's not typical.
